services/ui-src/src/components/App/App.js

```diff
@@ -23,14 +20,10 @@ function App() {
 
   async function onLoad() {
     try {
-      const user = (await Auth.currentSession()).getIdToken();
-      // *** make sure attributes exist and are in standard format
-      user.attributes = user.payload;
-      user.attributes["app-role"] = await determineRole(
-        user.attributes["custom:ismemberof"]
-      );
-
-      await ascertainUserPresence(user);
+      const token = (await Auth.currentSession()).getIdToken();
+      const apiUser = await ensureUserExistsInApi(token.payload.email);
+      const user = { attributes: apiUser }; // ew
```

Comment:
ew?

Reply:
Just to explain myself here... I would expect a user object to resemble this:
```typescript
user = {
  username: "george",
  email: "george@example.com",
  roles: "state_user"
}
```

Instead, we have this:
```typescript
user = {
  attributes: {
    username: "george",
    email: "george@example.com",
    roles: "state_user"
  }
}
```

This is for no reason that I can tell, and it adds noise throughout ui-src whenever we need to access those attributes. I find it gross, but to fix it would mean a whole _nother_ raft of changes. Maybe someday.